### PR TITLE
1486 - Giving the capability for admin to change user access level for any user

### LINKED ISSUE #1486: Create ability for Admins to promote and demote Admins
### Overview
Right now the only avenue to promote a user to admin level is to ping someone with DB access and have them promote the account. Not ideal.

### Action Items
The EditUser accessible by an admin needs the ability to promote and demote admins. Here's the short list:

#### PR Needs
- [ ] Promote a user to an admin
- [ ] Remove a user's admin status
- [ ] Remove self admin status

This will require both front end and back end edits

### Front end edits:
- [ ] Create AdminStatus component rendered on the EditUsers page 
  - [ ] Displays text: "VRMS Admin"  and toggle component.
```js
// client/src/components/user-admin/EditUsers.js
<FormGroup>
  <FormControlLabel control={<Switch checked={isActive} />} onClick={() => handleSetIsActive()} />
</FormGroup>
```

- [ ] Create function within UserApiService:
  - [ ] UpdateUserAdminStatus
  This function edits the user's admin status ~and stores the the user id `_id` of the admin that promoted them.~

#### Back end edits:
- [ ] ~update `user.model.js` to have an `adminPromotedBy` property of type `string`~
- [ ] apply the update to the user model to across the db

### Resources 
This task is fairly similar to #1452 , so feel free to use that PR's code as a model for how to do this PR.
- This issue is part of this epic: #1737

### PR DESCRIPTION
**Pull request by ntrehan**
Fixes #1486 

### What changes did you make and why did you make them ?

  - Provided a toggle in the edit users screen granting an admin the capability to change the access level of a user
  - refactored CSS files to avoid code repitition
  - added restrictions on the model layer allowing accessLevel to be either of the 2 strings "user" and "admin"

### Screenshots of Proposed Changes Of The Website 

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image

</details>

### CODE REVIEW

**Comment by trillium** [comment]
image

🤣 Perct, this allows me to remove admin access from Josh

File: client/src/components/user-admin/EditUsers.jsx [hunk 1]
```diff
@@ -74,15 +81,24 @@ const EditUsers = ({ userToEdit, backToSearch, updateUserDb, projects, updateUse
         <div className="user-display-column-left">Email:</div>
         <div className="user-display-column-right">{userEmail}</div>
       </div>
-      <div className="ua-row is-active-flex">
-        <div className="user-is-active-column-left">Is Active:</div>
-        <div className="is-active-flex">
-          <span className="active-status">{isActive.toString()}</span>
+      <div className="ua-row toggle-flex">
```
Comment (trillium) [inline client/src/components/user-admin/EditUsers.jsx:84]: Solid class rename

File: client/src/pages/UserAdmin.jsx [hunk 2]
```diff
@@ -37,6 +37,15 @@ const UserAdmin = () => {
     }, [userApiService, fetchUsers]
   )
 
+  // Update user's access level (admin/user)
```
Comment (trillium) [inline client/src/pages/UserAdmin.jsx:40]: Thanks for following convention by keeping the bd calls in one spot here

**Review by trillium — APPROVED** [review]
This is a solid PR, thank you @ntrehan !

**Comment by JackHaeg** [comment]
@ntrehan Just noticed that there is a small spelling error in the screenshots - "_VRSM_ ADMIN" when it should be "VRMS ADMIN"
Screenshot 2024-09-24 at 12 09 46 PM

**Pull request merged**